autorest/codegen/__init__.py

```diff
@@ -265,7 +265,7 @@ def _build_code_model_options(self) -> Dict[str, Any]:
             "package_name": self._autorestapi.get_value("package-name"),
             "package_version": self._autorestapi.get_value("package-version"),
             "client_side_validation": self._autorestapi.get_boolean_value("client-side-validation", False),
-            "tracing": self._autorestapi.get_boolean_value("trace", False),
+            "tracing": self._autorestapi.get_boolean_value("trace", version_tolerant),
```

Comment:
NOTES:
1. Only defaulting to `True` for `version-tolerant`, at least for now. We don't currently don't have tracing ability for client + request builders, so when / if we decide to do that, it should be additive to turn the flag to `True` then
2. No change in outputted generation, bc we generate with trace flag `True`. Checked that the default is true for version tolerant, but we have tests for tracing + legacy, so don't want to change that part of the test suite